keras_nlp/layers/preprocessing/mlm_mask_generator.py

```diff
@@ -0,0 +1,216 @@
+        random_token_rate: float, defaults to 0.1. `random_token_rate` must be
+            between 0 and 1 which indicates how often a random token is
+            substituted for tokens selected for masking. Default is 0.1.
+            Note: mask_token_rate + random_token_rate <= 1.
+        padding_token_id: int, defaults to None. The id of padding token.
+
+    Input:
+        A 1D integer tensor of shape [sequence_length,] or a 2D integer tensor
+        of shape [batch_size, sequence_length], or a 2D integer RaggedTensor.
+        Represents the sequence to mask.
+
+    Returns:
+        A Dict with 4 keys:
+            tokens: Tensor, has the same type and shape of input.
+                Sequence after getting masked.
+            mask_positions: Tensor, or RaggedTensor if
+                `mask_selection_length` is None. The positions of tokens
+                getting masked.
+            mask_ids: Tensor, or RaggedTensor if
+                `mask_selection_length` is None. The original token ids
+                at masked positions.
+            mask_weights: Tensor, or RaggedTensor if
+                `mask_selection_length` is None. `mask_weights` has the same
+                shape as `mask_positions` and `mask_ids`. Each element in
+                `mask_weights` should be 0 or 1, 1 means the corresponding
+                position in `mask_positions` is an actual mask, 0 means it is
+                a pad.
+
+    Examples:
+
+    Basic usage.
+    >>> masker = keras_nlp.preprocessing.MLMMaskGenerator( \
+    ...     10, mask_selection_rate=0.2, mask_selection_length=5)
+    >>> masker(tf.constant([1, 2, 3, 4, 5]))
+    {'tokens': <tf.Tensor: shape=(5,), dtype=int32,
+        numpy=array([1, 2, 3, 4, 0], dtype=int32)>,
+    'mask_positions': <tf.Tensor: shape=(5,), dtype=int64,
+        numpy=array([4, 0, 0, 0, 0])>,
+    'mask_ids': <tf.Tensor: shape=(5,), dtype=int32,
+        numpy=array([5, 0, 0, 0, 0], dtype=int32)>,
+    'mask_weights': <tf.Tensor: shape=(1, 5), dtype=int64,
+        numpy=array([[1, 0, 0, 0, 0]])>}
+
+    Ragged Input:
+    >>> masker = keras_nlp.preprocessing.MLMMaskGenerator( \
+    ...     10, mask_selection_rate=0.5, mask_selection_length=5)
+    >>> masker(tf.ragged.constant([[1, 2], [1, 2, 3, 4]]))
+    {'tokens': <tf.RaggedTensor [[1, 4], [0, 2, 3, 0]]>,
+    'mask_positions': <tf.Tensor: shape=(2, 5), dtype=int64, numpy=
+        array([[1, 0, 0, 0, 0],
+               [0, 3, 0, 0, 0]])>,
+    'mask_ids': <tf.Tensor: shape=(2, 5), dtype=int32, numpy=
+        array([[2, 0, 0, 0, 0],
+               [1, 4, 0, 0, 0]], dtype=int32)>,
+    'mask_weights': <tf.Tensor: shape=(2, 5), dtype=int64, numpy=
+        array([[1, 0, 0, 0, 0],
+               [1, 1, 0, 0, 0]])>}
+    """
+
+    def __init__(
+        self,
+        vocabulary_size,
+        mask_selection_rate,
+        mask_selection_length=None,
+        unselectable_token_ids=None,
+        mask_token_id=0,
```

Comment:
I'm not sure we can default this to 0, for most projects this is something different than padding token I think.

E.g.
BERT: 103
ROBERTA: 50264

We might want to make this required.

Reply:
sg, done